components/up-job.libsonnet

```diff
@@ -146,7 +150,63 @@ local k = import 'ksonnet/ksonnet.beta.4/k.libsonnet';
     ),
   },
 
+  withLogsFile:: {
+    local job = k.batch.v1.job,
+    local container = job.mixin.spec.template.spec.containersType,
+    local u = self,
+    config+:: {
+      bashImage: error 'must provide image for bash',
+    },
+
+    job+: {
+      spec+: {
+        template+: {
+          spec+: {
+            local c =
+              container.new('logs-file', u.config.bashImage) +
+              container.withCommand([
+                '/bin/sh',
+                '-c',
+                |||
+                  cat > /var/shared/logs.yaml << EOF
+                  spec:
+                    logs: [ [ "$(date '+%s%N')", "log line"] ]
+                  EOF
+                |||,
+              ]) +
+              container.withVolumeMounts({
+                name: 'shared',
+                mountPath: '/var/shared',
+                readOnly: false,
+              }),
+
+            initContainers+: [c],
+
+            containers: [
+              if c.name == 'observatorium-up' then c {
+                resources: u.config.resources,
+                args+: [
+                  '--logs-file=/var/shared/logs.yaml',
+                ],
+              } + container.withVolumeMounts({
+                name: 'shared',
+                mountPath: '/var/shared',
+                readOnly: false,
```

Comment:
this can be `readOnly: true` for this container, no?

Reply:
I addressed this point. You are absolutely right!